Striker/Striker/Striker_logic.ino

```diff
@@ -0,0 +1,141 @@
+Motors myMotors(
+    MOTOR1_PWM, MOTOR1_IN1, MOTOR1_IN2,
+    MOTOR2_PWM, MOTOR2_IN1, MOTOR2_IN2,
+    MOTOR3_PWM, MOTOR3_IN1, MOTOR3_IN2,
+    MOTOR4_PWM, MOTOR4_IN1, MOTOR4_IN2);
+
+float bno_angle = 0;
+float ball_angle = 0;
+float ball_distance = 0;
+float goal_angle = 0;
+float goal_distance = 0;
+float ball_angle_180 = 0;
+bool ball_found = false;
+bool has_ball = false;
+bool goal_found = false;
+double ponderated_angle = 0;
+double target_angle = 0;
+double last_time_ball_seen = 0;
+double time_threshold = 50;
+
+void setup()
+{
+    myMotors.InitializeMotors();
+    Serial.begin(115200);
+    while (!Serial && millis() < 10000UL)
+        ;
+    Serial.println("Started");
+    Serial1.begin(9600);
+}
+
+void loop()
+{
+    bno_angle = serialComm.Receive(RECEIVE_BNO);
+    ball_angle = serialComm.Receive(RECEIVE_BALL_ANGLE);
+    ball_distance = serialComm.Receive(RECEIVE_BALL_DISTANCE);
+    goal_angle = serialComm.Receive(RECEIVE_GOAL_ANGLE);
+    goal_distance = serialComm.Receive(RECEIVE_GOAL_DISTANCE);
+
+    double speed_w = pid_w.Calculate(target_angle, bno_angle);
+    double speed_t_goal = pid_t_goal.Calculate(180, goal_angle);
+    double speed_t_ball = pid_t_ball.Calculate(0, ball_distance);
+
+//--------------------------------------- Separate coordinate plane ---------------------------------//
+
+    if (ball_angle < 180)
+    {
+        ball_angle_180 = -ball_angle;
+    }
+    else if (ball_angle > 180)
+    {
+        ball_angle_180 = 360 - ball_angle;
+    }
+
+//--------------------------------------- Ball found logic ---------------------------------------//
+
+    if (ball_angle == 0)
+    {
+        ball_found = false;
+    }
+    else
+    {
+        ball_found = true;
+        last_time_ball_seen = millis();
+        Serial.println("Ball found");
+    }
+
+//--------------------------------------- Goal found logic ---------------------------------------//
+
+    if(goal_angle == 0)
+    {
+        goal_found = false;
+    }
+    else
+    {
+        goal_found = true;
+        //Serial.println("GOAL found");
+    }
+
+//--------------------------------------- Has ball logic ----------------------------------------//
+
+    if (ball_found && ball_distance < 20 && millis() - last_time_ball_seen < time_threshold && ball_angle_180 > -20 && ball_angle_180 < 20)
+    {
+        has_ball = true;
+        Serial.println("HAS BAALLL");
+    }
+    else
+    {
+        has_ball = false;
+    }
+
+//--------------------------------------- Scoring goal logic -------------------------------------//
+
+    if(has_ball && goal_found && goal_distance > 30)
+    {
+        // move towards goal
+        myMotors.MoveMotorsImu(0, abs(speed_t_ball), speed_w);
+    }//peruana
+    else if(has_ball && goal_found && goal_distance < 30)
+    {
+        // shoot
+        myMotors.MoveMotorsImu(360 - goal_angle, abs(speed_t_ball), speed_w);
+    }
+    else if (has_ball && !goal_found)
+    {
+        // logica de no tener porteria
+    }
```

Comment:
For the tests of the logic and due to the timing, I think it's okay to handle this using conditionals but maybe it would look cleaner in your code to implement a state machine, for example:
```cpp
enum States{
    Idle = 0,
    GoalKeeperPosition = 1,
    FollowBall = 2,
    ScoreGoal = 3
    ...
} current_state;

if (x_condition && y_condition)
    current_state = States::FollowBall;

switch(current_state){
    case FollowBall:
        // execution
    break;
    ...
```